### PR TITLE
SSH configuration test coverage

### PR DESCRIPTION
**Pull request by Monkeychip**
- [x] enterprise tests pass

### Description
This PR cleans up and adds test coverage for the SSH secret engine, focusing on the Configuration piece.

**Why**
- The WIF work involves refactoring SSH configuration routes and components. The intention however is that no user facing changes occur to the SSH engine. By fixing up the tests ahead of time, we can make sure that the upcoming changes behave in the older code as the do within the refactored code.

**Notable happenings**
- there were 2 ssh acceptance tests, one called `role-test` and one `ssh-test`. They both covered OTP role creation, deletion and generating keys. I combined them to make one `ssh/roles-test`, which covers ca roles and otp roles. There remains duplication that is very difficult to remove. If we ever spend time on this engine we can address this then, but what's been done is an improvement even though duplication remains (just less than before).

No user facing changes.

### CODE REVIEW

File: ui/tests/acceptance/secrets/backend/configuration/edit-test.js
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/configuration/edit-test.js:1]: this test checked configuration, all that coverage is now in the configuration-test.js file.

File: ui/tests/acceptance/secrets/backend/ssh/role-test.js [hunk 1]
```diff
@@ -3,110 +3,221 @@
...
-    );
-
-    await listPage.visitRoot({ backend: path });
-    await settled();
-    assert.strictEqual(listPage.secrets.length, 1, 'shows role in the list');
-    const secret = listPage.secrets.objectAt(0);
-    await secret.menuToggle();
-    assert.dom('.hds-dropdown li').exists({ count: 5 }, 'Renders 5 popup menu items');
-  });
+  const PUB_KEY = `ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCn9p5dHNr4aU4R2W7ln+efzO5N2Cdv/SXk6zbCcvhWcblWMjkXf802B0PbKvf6cJIzM/Xalb3qz1cK+UUjCSEAQWefk6YmfzbOikfc5EHaSKUqDdE+HlsGPvl42rjCr28qYfuYh031YfwEQGEAIEypo7OyAj+38NLbHAQxDxuaReee1YCOV5rqWGtEgl2VtP5kG+QEBza4ZfeglS85f/GGTvZC4Jq1GX+wgmFxIPnd6/mUXa4ecoR0QMfOAzzvPm4ajcNCQORfHLQKAcmiBYMiyQJoU+fYpi9CJGT1jWTmR99yBkrSg6yitI2qqXyrpwAbhNGrM0Fw0WpWxh66N9Xp meirish@Macintosh-3.local`;
```
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/ssh/role-test.js:32]: This was here before, not added. Might help to compare to current [ssh-test file](https://github.com/hashicorp/vault/blob/main/ui/tests/acceptance/secrets/backend/ssh/ssh-test.js) as the diff is a little hard to parse.

File: ui/tests/acceptance/secrets/backend/ssh/roles-test.js
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/ssh/roles-test.js:1]: The base of this new test was [ssh-test](https://github.com/hashicorp/vault/blob/main/ui/tests/acceptance/secrets/backend/ssh/ssh-test.js) and added to it under the model(opt roles) was [role-test](https://github.com/hashicorp/vault/blob/main/ui/tests/acceptance/secrets/backend/ssh/role-test.js). There was one easy point of duplication I removed which was under the role-test the test "c[reate and redirects](https://github.com/hashicorp/vault/blob/main/ui/tests/acceptance/secrets/backend/ssh/role-test.js#L35)". this was covered in the first test here, "it creates roles, generates keys and deletes roles". So this was the only chunk removed. The rest was combined, opting for duplication over missing previous coverage.

File: ui/tests/acceptance/secrets/backend/configuration/edit-test.js
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/configuration/edit-test.js:1]: This test covered the configuration flow, which is covered in the configuration-test.js

File: ui/tests/acceptance/secrets/backend/ssh/roles-test.js
Comment (Monkeychip) [reply ui/tests/acceptance/secrets/backend/ssh/roles-test.js:1]: If folks think that I should not touch roles things at all, I can leave this as was. I was doing a selector sweep of all ssh tests and realized the duplication and confusing test names.

File: ui/app/components/secret-engine/configure-ssh.hbs [hunk 2]
```diff
@@ -16,7 +16,7 @@
           @value={{@model.publicKey}}
           @displayOnly={{true}}
           @allowCopy={{true}}
-          data-test-ssh-input="public-key"
```
Comment (Monkeychip) [inline ui/app/components/secret-engine/configure-ssh.hbs:19]: looking at other areas, it appears that we tend to _not_  include the engine-type in the selector name. Removing 'ssh' here for consistency.

File: ui/tests/acceptance/secrets/backend/ssh/configuration-test.js [hunk 3]
```diff
@@ -61,32 +66,61 @@ module('Acceptance | ssh | configuration', function (hooks) {
...
+    const path = `ssh-configure-${this.uid}`;
+    await enablePage.enable('ssh', path);
+    await click(SES.configTab);
+    await click(SES.configure);
+    assert
+      .dom(GENERAL.inputByAttr('generate-signing-key-checkbox'))
+      .isChecked('generate_signing_key defaults to true');
+    await click(GENERAL.inputByAttr('generate-signing-key-checkbox'));
+    await click(SES.ssh.save);
+    assert.true(this.flashDangerSpy.calledWith('missing public_key'), 'Danger flash message is displayed');
```
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/ssh/configuration-test.js:115]: This shouldn't be a flash message but an actual error on the page. Stay tuned, it becomes one in the ssh refactor.

File: ui/tests/integration/components/secret-engine/configure-ssh-test.js
Comment (Monkeychip) [inline ui/tests/integration/components/secret-engine/configure-ssh-test.js:1]: new component tests focusing on component only things. This will change a lot in the next SSH refactor pr, but it will be helpful to see exactly how the component's responsibilities are changing.

File: ui/tests/acceptance/secrets/backend/ssh/roles-test.js
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/ssh/roles-test.js:1]: The base of this new test was [ssh-test](https://github.com/hashicorp/vault/blob/main/ui/tests/acceptance/secrets/backend/ssh/ssh-test.js). Added to that base is the old [role-test](https://github.com/hashicorp/vault/blob/main/ui/tests/acceptance/secrets/backend/ssh/role-test.js). These two test were testing the same thing in different ways. These are combined copies of the two test, but for one clear duplication: under the old role-test's "c[reate and redirects](https://github.com/hashicorp/vault/blob/main/ui/tests/acceptance/secrets/backend/ssh/role-test.js#L35)" was removed as it's covered in the first test below, "it creates roles, generates keys and deletes roles".  

If folks think it's best for me not to touch anything roles ssh related, I'm happy to leave as was. I was doing an overall ssh sweep for test selectors and noticed the duplication and slightly confusing naming so thought I would address.

File: ui/tests/acceptance/secrets/backend/configuration/edit-test.js
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/configuration/edit-test.js:1]: This test covered the configuration flow, which is covered in the configuration-test.js

**Review by hashishaw — APPROVED** [review]
Thank you for splitting this out!!

File: ui/app/components/secret-engine/configure-ssh.hbs [hunk 4]
```diff
@@ -34,11 +34,12 @@
         @buttonColor="secondary"
         @confirmMessage="This will remove the CA certificate information."
         @onConfirmAction={{this.delete}}
+        data-test-delete-public-key
       />
     </Hds::ButtonSet>
   </div>
 {{else}}
-  <form {{on "submit" this.saveConfig}} data-test-ssh-configure-form="true">
+  <form {{on "submit" this.saveConfig}} data-test-configure-form="true">
```
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/app/components/secret-engine/configure-ssh.hbs:42]: Good catch!
